### ISSUE TITLE
mport search sometimes segfaulting

### ISSUE THREAD

**Issue by laffer1** [MidnightBSD/mport#47]
A user reported an issue with mport search while looking for the drm-kmod package.

mport search drm-kmod works. mport search drm seems to crash.

Figure out what's going on

**Comment by laffer1** [comment]
```
Starting program: /usr/sbin/mport search drm

Program received signal SIGSEGV, Segmentation fault.
0x0000000801b64ae5 in sqlite3_str_vappendf (pAccum=<value optimized out>, fmt=0x404928 "Q or comment glob %Q", 
    ap=<value optimized out>) at /usr/src/lib/libsqlite3/../../contrib/sqlite3/sqlite3.c:30251
30251	        for(i=n=0; k!=0 && (ch=escarg[i])!=0; i++, k--){
Current language:  auto; currently minimal
(gdb) bt
#0  0x0000000801b64ae5 in sqlite3_str_vappendf (pAccum=<value optimized out>, 
    fmt=0x404928 "Q or comment glob %Q", ap=<value optimized out>)
    at /usr/src/lib/libsqlite3/../../contrib/sqlite3/sqlite3.c:30251
#1  0x0000000801b67191 in sqlite3_vmprintf (zFormat=0x40491e "pkg glob %Q or comment glob %Q", ap=0x7fffffffe8b0)
    at /usr/src/lib/libsqlite3/../../contrib/sqlite3/sqlite3.c:30676
#2  0x000000080083b703 in mport_index_search (mport=0x803230000, entry_vec=0x7fffffffe930, 
    fmt=<value optimized out>) at /usr/src/contrib/mport/libmport/index.c:454
#3  0x00000000004037d3 in search (mport=0x803230000, query=0x803235010)
    at /usr/src/contrib/mport/mport/mport.c:454
#4  0x00000000004027f8 in main (argc=2, argv=0x7fffffffeb78) at /usr/src/contrib/mport/mport/mport.c:273
#5  0x0000000000401a7b in _start ()
#6  0x000000080062a000 in ?? ()
#7  0x0000000000000000 in ?? ()
```

**Comment by laffer1** [comment]
fixed in 9be5550d5ac08639c04e3b4f5fad2c3ba81f01da
